### PR TITLE
WIP Misuv 5188: Add SE | Build - Business address look up

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: app/config/AddressLookupConfig.scala [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import uk.gov.hmrc.hmrcfrontend.config.AccessibilityStatementConfig
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class AddressLookupConfig @Inject()(appConfig: FrontendAppConfig, messagesApi: MessagesApi, accessibilityStatementConfig: AccessibilityStatementConfig){
+
+  def config(continueUrl: String)(implicit request: RequestHeader): JsObject =
+
+    Json.obj (
```
Comment (avelx) [inline app/config/AddressLookupConfig.scala:31]: you will have to pass English and Welsh resources in some why if you want to prepare this config in the scope this method

File: app/connectors/AddressLookupConnector.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  lazy val individualContinueUrl: String = routes.AddBusinessAddressController.submit(None).url
+  lazy val agentContinueUrl: String = routes.AddBusinessAddressController.agentSubmit(None).url
+
+
+  //"http://localhost:9081/report-quarterly/income-and-expenses/view/income-sources/add/business-check-details"
+  def addressJson(continueUrl: String): JsValue = {
+    play.api.libs.json.Json.parse(
+      s"""
```
Comment (avelx) [inline app/connectors/AddressLookupConnector.scala:57]: I would suggest to generate this json using case classes + there are urls which need to be generate dynamically: i.e  http://localhost:9081/report-quarterly/income-and-expenses/view/sign-out etc

File: app/config/AddressLookupConfig.scala (hunk 1 above)
Comment (danielmaltz1) [reply app/config/AddressLookupConfig.scala]: This file was leftover from a previous attempt, config Json is now created in the connector, I'll add the english and welsh content once it has been confirmed by a BA

File: app/controllers/AddBusinessAddressController.scala [hunk 3]
```diff
@@ -16,31 +16,125 @@
...
+    if (isDisabled(IncomeSources)) {
+      Future.successful(Redirect(controllers.routes.HomeController.show()))
+    } else {
+        addressLookupConnector.initialiseAddressLookup(
+          isAgent = isAgent
+        ) map {
+          case Right(PostAddressLookupSuccessResponse(Some(location))) =>
+            Redirect(location)
+          case Right(PostAddressLookupSuccessResponse(None)) =>
+            throw new InternalServerException(s"[AddBusinessAddressController][handleRequest] - Unexpected response, success, but no location returned")
```
[the remark this answers is not in the document]
Comment (danielmaltz1) [reply app/controllers/AddBusinessAddressController.scala]: I've tried to change this in the controller, but I'm hitting strange errors, will keep working on it

File: test/connectors/AddressLookupConnectorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      "return the get url" in {
+        disableAllSwitches()
+        enable(IncomeSources)
+
+        val result = TestAddressLookupConnector.getAddressDetailsUrl("123")
+        result shouldBe s"${baseUrl}/api/v2/confirmed?id=123"
+      }
+    }
+
+    /*"initialiseAddressLookup" should {
```
Comment (avelx) [inline test/connectors/AddressLookupConnectorSpec.scala:63]: remove not used code

**Pull request merged**